plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractTrinoCatalog.java

```diff
@@ -221,6 +240,35 @@ protected void deleteTableDirectory(TrinoFileSystem fileSystem, SchemaTableName
         }
     }
 
+    protected String createMaterializedViewStorage(ConnectorSession session, SchemaTableName viewName, ConnectorMaterializedViewDefinition definition)
```

Comment:
If you pass also `fileSystemFactory` to this method, there is no need to pass anymore this field as a parameter to the class's constructor.